pylint/checkers/utils.py

```diff
@@ -2032,35 +2031,33 @@ def is_augmented_assign(node: nodes.Assign) -> tuple[bool, str]:
     return False, ""
 
 
+def _qualified_name_parts(module_name: str) -> list[str]:
+    """Split the names of the given module into subparts.
+
+    For example,
+        _qualified_name_parts('pylint.checkers.ImportsChecker')
+    returns
+        ['pylint', 'pylint.checkers', 'pylint.checkers.ImportsChecker']
+    """
+    names = module_name.split(".")
+    return [".".join(names[0 : i + 1]) for i in range(len(names))]
+
+
 def is_module_ignored(
-    module: nodes.Module,
+    module_name: str,
```

Comment:
```suggestion
    qualified_module_name: str,
```
```suggestion
    module: nodes.Module,
```
I think the refactor to give a string instead of a node module might not be the best in term of potential error being made later. It's possible to forget to give the qualified name by mistake (the name is still a string), but if you have to give a ``node.Module``, mypy will remind you that. Also it means the complexity of handling the module ignore is is kept inside the function that deals with it.

Reply:
I agree, but the function also needs to be used by `_ignore_import_failure` (called by `_get_imported_module`), which only has access to an `ImportNode` and a string, which is the qualified module name. I don't think it's possible to create a `nodes.Module` without actually resolving the import, which we can't do in the case of `import-error`. Or do you have a different suggestion?

Reply:
We should probably just rename the str argument to ``qualified_module_name`` like in the first suggestion then.